crypto/evp/evp_enc.c

```diff
@@ -1546,8 +1548,11 @@ void EVP_CIPHER_free(EVP_CIPHER *cipher)
 {
     int i;
 
-    if (cipher == NULL)
+    if (cipher == NULL || cipher->origin == ORIG_GLOBAL)
         return;
+    if (cipher->origin == ORIG_METH) {
+        /* EVP_CIPHER_meth_free just calls this function. */
```

Comment:
Same comment as above

Reply:
I like explicitly handling all the cases.  (I'd use an enum if it were more like OpenSSL style and let the compiler warn about unhandled things.)